shop4All-backend/src/main/java/com/example/shop4All_backend/Configuration/WebSecurityConfiguration.java

```diff
@@ -1,30 +1,84 @@
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.web.cors.CorsConfigurationSource;
 
 import static org.springframework.security.config.Customizer.withDefaults;
 
+
+
 @Configuration
 @EnableWebSecurity
+@EnableMethodSecurity(securedEnabled = true)
 public class WebSecurityConfiguration {
 
+    @Autowired
+    private JwtAuthentificationEntryPoint jwtAuthentificationEntryPoint;
+
+    @Autowired
+    private JwtRequestFilter jwtRequestFilter;
+
+    @Autowired
+    private UserDetailsService jwtService;
+
+    // Define the authentication manager bean
     @Bean
-    public SecurityFilterChain configureOauth2(HttpSecurity http) throws Exception {
+    public AuthenticationManager authenticationManagerBean(AuthenticationConfiguration authenticationConfiguration) throws Exception {
+        return authenticationConfiguration.getAuthenticationManager();
+    }
+
+    // Configure security settings including OAuth2 and JWT authentication
+    @Bean
+    public SecurityFilterChain configure(HttpSecurity http, CorsConfiguration corsConfiguration) throws Exception {
         return http
+                // Disable CSRF protection
+                .csrf(AbstractHttpConfigurer::disable)
+                // Configure authorization rules for different endpoints
                 .authorizeHttpRequests(auth -> {
                     auth.requestMatchers("/").permitAll();
-                    auth.requestMatchers("/register/buyer").permitAll();
-                    auth.requestMatchers("/register/seller").permitAll();
-                    auth.requestMatchers("/shop4All").permitAll();
-                    auth.requestMatchers("/shop4All/product/**").permitAll();
+                    auth.requestMatchers("/registerNewSeller").permitAll();
+                    auth.requestMatchers("/authenticate").permitAll();
+                    auth.requestMatchers("/login").permitAll();
                     auth.anyRequest().authenticated();
                 })
+                .exceptionHandling((exceptions) -> exceptions
+                        .authenticationEntryPoint(jwtAuthentificationEntryPoint)
+                )
+                .sessionManagement((session) -> session
+                        .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+                )
                 .formLogin(withDefaults())
```

Comment:
In the future, you won't use the login form provided by Spring Security so you can delete this line in a future PR or you can disable the login form.